client/layout/masterbar/style.scss

```
@@ -1401,3 +1414,53 @@ a.masterbar__quick-language-switcher {
 		opacity: 1;
 	}
 }
+
+.masterbar__item.has-global-border .item-border-wrapper {
+	position: absolute;
+	display: none;
+
+	.is-global-sidebar-visible &.is-active {
+		display: block;
+		top: 0;
+		bottom: 0;
+		right: 0;
+		left: 0;
+
+		.item-border-wrapper__left,
+		.item-border-wrapper__right {
+			display: block;
+			position: absolute;
+			height: var(--masterbar-item-active-border-radius);
+			width: var(--masterbar-item-active-border-radius);
+			background-color: var(--color-sidebar-background);
+			bottom: 0;
+			z-index: 1;
```

```suggestion
	display: block;
	opacity: 0;

	.is-global-sidebar-visible &.is-active {
		opacity: 1;
		top: 0;
		bottom: 0;
		right: 0;
		left: 0;

		.item-border-wrapper__left,
		.item-border-wrapper__right {
			display: block;
			position: absolute;
			height: var(--masterbar-item-active-border-radius);
			width: var(--masterbar-item-active-border-radius);
			background-color: var(--color-sidebar-background);
			bottom: 0;
			z-index: -1;
			transition: all 0.2s ease-in-out;
```
This is how it looks (first part of the video with animation at 10%, and then 100%)

https://github.com/user-attachments/assets/e9e0fcf9-543a-488c-9cdd-b41e99087b3f

---

Thanks for exploring this! I pushed a commit based on what you explored and some things you mentioned in slack. Its definitely a bit better than it was now!

I am going to try again to get this same outward curve effect with less elements, as that would probably have less potential visual conflicts like this.

---

> to try again to get this same outward curve effect with less elements, as that would probably have less potential visual conflicts like this.

that ended up working out! we can cut pseudo elements with a mask instead, removing the elements that were making the transition funny.